velox/functions/lib/SubscriptUtil.h

```diff
@@ -433,7 +434,8 @@ class SubscriptImpl : public exec::Subscript {
       bool found = false;
       auto searchIndex = searchIndices[row];
       for (auto i = 0; i < size; i++) {
-        if (mapKeysBase->equalValueAt(searchBase, offset + i, searchIndex)) {
+        if (mapKeysBase->equalValueAt(
```

Comment:
alternatively we can use mapKeys and offset + i.  and delete lines 416 and 417. 
  for (auto i = 0; i < size; i++) {
        if (mapKeys->equalValueAt(searchBase, offset + i, searchIndex)) 
        
 i assume this was done here like this for performance reasons? but its a virtual call anyway i will let you decide

Reply:
same apply to the indexArg, not sure why we are decoding it 
we could have done instead:
if (mapKeys->equalValueAt(indexArg.get(), offset + i, row))

Like we did not need to decode maps keys nor indices, its not clear to me that its  a saving

Reply:
This should be the same reason as I explained in your comments in `ArrayContains.cpp`.